lib/classes/record/adapter.php

```diff
@@ -31,18 +31,257 @@
  */
 class record_adapter implements record_Interface, cache_cacheableInterface
 {
+
+    public static function get(Application $app, $sbas_id, $record_id, $number = null)
+    {
+        return new self($app, $sbas_id, $record_id, $number, true);
+    }
+
+    public static function getMany(Application $app, $sbas_id, $record_ids = array())
+    {
+        $databox = $app['phraseanet.appbox']->get_databox((int) $sbas_id);
+        $connection = $databox->get_connection();
+
+        $ids = array_map(function ($id) {
+            if (is_array($id)) {
+                return (int) $id['id'];
+            }
+
+            return (int) $id;
+        }, $record_ids);
+
+        $numbers = array();
+
+        foreach ($record_ids as $id) {
+            $numbers[$id['id']] = null;
```

Comment:
use of `$id` as an array. You cannot be sure of this. You did normalize `$ids` for this, didn't you?